api/v3rpc/rpctypes/error.go

```diff
@@ -277,3 +279,25 @@ func ErrorDesc(err error) string {
 	}
 	return err.Error()
 }
+
+func ContextError(ctx context.Context, err error) error {
```

Comment:
```suggestion
// ConvertError converts the error into an EtcdError if the error message matches one of
// the defined messages; otherwise, it tries to retrieve the context error.
func ConvertError(ctx context.Context, err error) error {
```

Reply:
It's only used by the client side, so suggest to put it in `clientv3`. 

If you want quick merge, suggest to do it in a separate PR.